### ISSUE TITLE
`PendingInstall` and `PendingRemoval` apps not installed or removed during upgrades

### ISSUE THREAD

**Issue by pablofsf** [Cogitri/gnome-software-plugin-apk#16]
Those apps are totally skipped, generating inconsistencies. For example, a simple `sudo apk upgrade` installed and removed some packages after the system was in theory completely up-to-date according to the UI.

That is happening because currently apps are not marked for an upgrade unless state is `Upgradable` or `Downgradable`.

When adding this functionality, to produce the correct results in the UI, `PendingInstall` apps should be marked as `GS_APP_STATE_AVAILABLE` and `PendingRemove` apps as `GS_APP_STATE_UNAVAILABLE`.

**Comment by Cogitri** [comment]
Hm, I added PendingInstall and PendingRemoval since I had assumed apk would usually automatically deal with them upon upgrading all the other packages. If that's not always the case, we should handle them properly in GS

**Comment by pablofsf** [comment]
Yes, I do not understand the apk-rs-polkit yet enough to know why it is not the case. The fact that I can barely read rust does not help either...
